src/ScottPlot5/ScottPlot5/Plottables/Heatmap.cs

```diff
@@ -258,7 +269,23 @@ public double GetValue(Coordinates coordinates)
 
     public IEnumerable<LegendItem> LegendItems => Enumerable.Empty<LegendItem>();
 
-    public Range GetRange() => Range.GetRange(Intensities);
+    public Range GetRange()
+    {
+        if (UseForcedRange)
+        {
+            return ForcedRange;
+        }
+        if (IntensityRange == default(Range))
```

Comment:
This check doesn't catch if the heatmap has had its intensities changed through an `Update` call. It will only catch if a range has never been set (i.e. on first render).

Reply:
Worth noting that `GetRange` is only called in this class when an `Update` is requested and we need to invalidate the cached bitmap. So it's only the consumers outside of this class that would benefit, and I think the only one that would include is `ColorBar`, which provides colour axes.

If you're interested, `ColorBar` probably has some room for performance improvements since it doesn't currently even cache its image (which depends only on its colour map), nor its axis, which depends only on its range and configured edge. I imagine that caching the bitmap in particular would have bigger gains just because creating it involves some not insubstantial heap allocations. Including a pinned heap allocation at that, which I think the garbage collector doesn't like as much.

Reply:
I made `ManualRange` nullable and have it automatically call `Update()` in the setter. I think this avoids the concerns mentioned above...